unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/SearchRecordComponentView.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using TMPro;
+using UIComponents.Scripts.Components;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class SearchRecordComponentView : BaseComponentView<SearchRecordComponentModel>, ISearchRecordComponentView
+{
+    [SerializeField] internal TMP_Text recordText;
+    [SerializeField] internal GameObject historyIcon;
+    [SerializeField] private Button recordButton;
+    [SerializeField] internal GameObject playerCountParent;
+    [SerializeField] internal TMP_Text playerCount;
+
+    public event Action<string> OnSelectedHistoryRecord;
+    public event Action<string> OnSelectedRegularRecord;
+
+    public override void Awake()
+    {
+        base.Awake();
+
+        recordButton.onClick.RemoveAllListeners();
+        recordButton.onClick.AddListener(OnSelectedRecord);
```

Comment:
was it you who told me that you don't like RemoveAllListeners 😅?  it is ok (just remembered it)

Reply:
I think it was someone else, I like it :D